i2c-flash.c

```diff
@@ -22,12 +22,20 @@ ISR(TWI_vect) {
 
 	uint8_t offset;
 
-	switch (TWSR) {
+	switch (TW_STATUS) {
+	case TW_BUS_ERROR:
+		TWCR = _BV(TWEN) | _BV(TWIE) | _BV(TWEA) | _BV(TWINT) | _BV(TWSTO);
+		while (TWCR & _BV(TWSTO))
+			continue;
+	case TW_NO_INFO:
```

Comment:
The datasheet says that we should never observe this value after an interrupt. If we do, it's still probably more correct to clear the interrupt instead of returning immediately. So I'd remove this case.

Reply:
It may say that, but it did not work correctly until I copied this here from the Arduino IDE's library.